meson.build

```diff
@@ -0,0 +1,142 @@
+project('libproxy', 'c',
+          version: '0.5.0',
+    meson_version: '>= 0.59.0',
+  default_options: [ 'warning_level=2', 'werror=false', 'c_std=gnu11', ],
+)
+
+version_arr = meson.project_version().split('-')[0].split('.')
+libproxy_version_major = version_arr[0].to_int()
+libproxy_version_minor = version_arr[1].to_int()
+
+root_dir = include_directories('.')
+
+px_prefix = get_option('prefix')
+datadir = get_option('datadir')
+px_libdir = join_paths(px_prefix, get_option('libdir'))
+px_plugins_dir = join_paths(px_libdir, meson.project_name().to_lower(), 'plugins')
+girdir     = get_option('datadir') / 'gir-1.0'
+typelibdir = get_option('libdir')  / 'girepository-1.0'
+vapidir = get_option('datadir') / 'vala' / 'vapi'
+
+add_project_arguments(['-I' + meson.project_build_root()], language: 'c')
+
+# The major api version as encoded in the libraries name
+api_version = '1.0'
+package_api_name = '@0@-@1@'.format(meson.project_name(), api_version)
+
+cc = meson.get_compiler('c')
+
+project_c_args = []
+test_c_args = [
+  '-Wcast-align',
+  '-Wdeclaration-after-statement',
+  '-Werror=address',
+  '-Werror=array-bounds',
+  '-Werror=empty-body',
+  '-Werror=implicit',
+  '-Werror=implicit-function-declaration',
+  '-Werror=incompatible-pointer-types',
+  '-Werror=init-self',
+  '-Werror=int-conversion',
+  '-Werror=int-to-pointer-cast',
+  '-Werror=main',
+  '-Werror=misleading-indentation',
+  '-Werror=missing-braces',
+  '-Werror=missing-include-dirs',
+  '-Werror=nonnull',
+  '-Werror=overflow',
+  '-Werror=parenthesis',
+  '-Werror=pointer-arith',
+  '-Werror=pointer-to-int-cast',
+  '-Werror=redundant-decls',
+  '-Werror=return-type',
+  '-Werror=sequence-point',
+  '-Werror=shadow',
+  '-Werror=strict-prototypes',
+  '-Werror=trigraphs',
+  '-Werror=undef',
+  '-Werror=write-strings',
+  '-Wformat-nonliteral',
+  '-Wignored-qualifiers',
+  '-Wimplicit-function-declaration',
+  '-Wlogical-op',
+  '-Wmissing-declarations',
+  '-Wmissing-format-attribute',
+  '-Wmissing-include-dirs',
+  '-Wmissing-noreturn',
+  '-Wnested-externs',
+  '-Wno-cast-function-type',
+  '-Wno-dangling-pointer',
+  '-Wno-missing-field-initializers',
+  '-Wno-sign-compare',
+  '-Wno-unused-parameter',
+  '-Wold-style-definition',
+  '-Wpointer-arith',
+  '-Wredundant-decls',
+  '-Wstrict-prototypes',
+  '-Wswitch-default',
+  '-Wswitch-enum',
+  '-Wundef',
+  '-Wuninitialized',
+  '-Wunused',
+  '-fno-strict-aliasing',
+  ['-Werror=format-security', '-Werror=format=2'],
```

Comment:
Dang... you already set warning level 2 (-Wextra); isn't that enough? Did you really think about which warnings to enable/disable or did you just copy/paste this from somewhere? Especially weird that this list is used only for tests. I doubt tests require -fno-strict-aliasing, for example.

Reply:
That's based on GNOME Builder template for a new project.

Reply:
I think you got it wrong. These are not test args for the tests. They are checked whether the compiler supports them and if yes, they are added to project args.

Reply:
OK, good.

Anyway, it's your project, so if you want to use these flags, then fine. But it looks like this list was clearly not designed for use with -Wextra, which you're also getting via warning level 2.